pkg/segment/query/processor/searcher.go

```diff
@@ -557,45 +594,76 @@ func (s *Searcher) getFilteredBlocks(blocks []*block) []*block {
 	return filteredBlocks
 }
 
-func (s *Searcher) getQSRSToProcess() []*query.QuerySegmentRequest {
-	qsrs := make([]*query.QuerySegmentRequest, 0)
+func (s *Searcher) shouldProcessQSR(cutOff uint64, qsr *query.QuerySegmentRequest) bool {
+	switch s.sortMode {
+	case recentFirst:
+		return qsr.GetEndEpochMs() >= cutOff
```

Comment:
nit: here and in `willProcessQSRCompletely()` you pass `cutOff` as a parameter, but in `shouldProcessBlock()` you use `s.cutOffTimestamp`. I'd suggest doing it the same way in each of these functions, unless there's a good reason not to

Reply:
addressed in the latest commit.